src/battle_script_commands.c

```diff
@@ -17209,3 +17209,12 @@ void BS_FickleBeamDamageCalculation(void)
         gBattlescriptCurrInstr = cmd->nextInstr;
     }
 }
+
+void BS_JumpIfAllyBlockSoundMove(void)
```

Comment:
Rename would apply here too.

Reply:
it specifically blocks ally.

Reply:
It doesn't need to block anything else because this is already handled elsewhere.

Reply:
Renaming would be misleading because it should not be used outside of ally

Reply:
Then why receive the battler argument if it's only going to use `BS_ATTACKER_PARTNER`?

Reply:
Because it isn't a targeting move. We are using a script to hardcode the value of the partner in the first place. The reason we need to make an exceptions for this move with a script in the first place is because we have no way of handling it in C. I've made a suggestion how it could be solved in the issue ticket but wasn't able to do it.

Reply:
Perish Song is not a targetting move either but could benefit from this change as well, as they use `BS_TARGET` for looping through all battlers, and therefore would not be partner-exclusive.
Without this, if a custom move were to use Perish Song's effect but without the Sound flag, Soundproof would still block it.

Reply:
Addendum: even ignoring this other use, I still believe that a function should be called based on what it does instead of how it's used :)

Reply:
how is it not named based on what it does?

Reply:
> Perish Song is not a targetting move either but could benefit from this change as well, as they use `BS_TARGET` for looping through all battlers, and therefore would not be partner-exclusive. Without this, if a custom move were to use Perish Song's effect but without the Sound flag, Soundproof would still block it.

addressed

Reply:
Because the name `JumpIfAllyBlockSoundMove` with `battler` as an argument makes me think that the *ally* of the battler being passed is the one that's being checked for Soundproof instead of the battler itself like how the function actually works.

Reply:
> addressed

It was good before, it just needed to be applied to Perish Song's script.

Reply:
That's a cleaner solution which avoids a redundant function.

Reply:
> That's a cleaner solution which avoids a redundant function.

But it muddles what `jumpifability` does.

Reply:
why? I would actually need to copy a lot of the code from `jumpifability` which I previously didn't realize, specifically the recordingpart and lastused ability. Generally if there is a function that does the same thing which should avoid duplication

Reply:
Because what if you need to use `jumpifability` for checking Soundproof something that isn't related to cancelling?

Eg. a custom sound move that damages but sets a flag only for Pokémon with Soundproof.

Reply:
> Because what if you need to use `jumpifability` for checking Soundproof something that isn't related to cancelling?
> 
> Eg. a custom sound move that damages but sets a flag only for Pokémon with Soundproof.

That isn't how multi target moves work. Each hit is done individually for each battler. You can see it when you look at a recording of a spread damage move. This also means that each script is run individually.

That is why it isn't that easy to make simultaneous hp reduction work. 

Anyways I brought the callnative back